### PR TITLE
DDS namespace updates

### PR DESCRIPTION
**Pull request by adamsj-ros**
- differences found versus current version of this repos's source and other build errors
- other likely candidates (i.e. DDS_)

### CODE REVIEW

**Comment by adamsj-ros** [comment]
@mitza-oci  and @neeleym, I found these DDS namespace changes from the rmw_connext. I think most of these are correct with the possible exception of DDS::Long and DDS::Octet.

**Review by neeleym — CHANGES_REQUESTED** [review]
DDS::Long should be CORBA::Long
DDS::Octet should be CORBA::Octet
DDS::String_free should be CORBA::string_free

**Comment by adamsj-ros** [comment]
> DDS::Octet should be CORBA::Octet

Is this true for IDL files too?

File: rmw_opendds_cpp/src/rmw_publisher.cpp [hunk 1]
```diff
@@ -195,20 +195,20 @@ rmw_create_publisher(
 
     topic = participant->create_topic(
       topic_str, type_name.c_str(),
-      default_topic_qos, NULL, DDS_STATUS_MASK_NONE);
+      default_topic_qos, NULL, DDS::STATUS_MASK_NONE);
```
Comment (jwillemsen) [inline rmw_opendds_cpp/src/rmw_publisher.cpp:198]: See https://github.com/objectcomputing/OpenDDS/issues/1306, this mask is not yet there in OpenDDS

Comment (adamsj-ros) [reply rmw_opendds_cpp/src/rmw_publisher.cpp:198]: Good find. At this point, I'm only trying to make some changes that are mostly cosmetic. @neeleym is helping me review the real OpenDDS interface. I see that you're correct. We'll add this to our list to fix.

**Pull request merged**